mods/persistence/modules/projectiles/overrides.dm

```diff
@@ -47,4 +47,22 @@
 	research_excluded = TRUE
 
 /datum/fabricator_recipe/protolathe/ammo/ammo_emp_slug
-	research_excluded = TRUE
\ No newline at end of file
+	research_excluded = TRUE
+
+/obj/item/gun/projectile/update_base_icon() // sets streamlined way for mag-fed guns to check they sprites
```

Comment:
Won't this cause issues with subtypes of ``/obj/item/gun/projectile`` that don't have these additional icon states, or do all the guns we current have have them?

Reply:
I was worried about this, but when I tested it all other weapon subtypes (currently that'd be speedloaders, individually loaded, and bolt-action) have their own 'update_base_icon' up the line, so it doesn't appear to be breaking anything that I could tell when I was testing it.

That's also why the second 'pistol' update_base_icon is needed, because otherwise pistols have their own update_base_icon that overrides this one.